### PR TITLE
feat: sso login with domain open login webpage

### PR DESCRIPTION
**Pull request by MohamadJaara**
----
#### PR Submission Checklist for internal contributors

- The **PR Title**
    - [ ] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
    - [ ] contains a reference JIRA issue number like `SQPIT-764`
    - [ ] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
    - [ ] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

### Issues

fetch user default sso code and start sso flow 

### Solutions

after user accepts backend switch check if the new server have a default sso code, if so then start sso login flow to the new server

### Dependencies (Optional)

_If there are some other pull requests related to this one (e.g. new releases of frameworks), specify them here._

Needs releases with:

- [ ] GitHub link to other pull request

#### Test Coverage (Optional)

- [ ] I have added automated test to this contribution

#### How to Test

_Briefly describe how this change was tested and if applicable the exact steps taken to verify that it works as expected._

### Notes (Optional)

_Specify here any other facts that you think are important for this issue._

### Attachments (Optional)

_Attachments like images, videos, etc. (drag and drop in the text box)_ 

----
#### PR Post Submission Checklist for internal contributors (Optional)

- [ ] Wire's Github Workflow has automatically linked the PR to a JIRA issue
----
#### PR Post Merge Checklist for internal contributors

- [ ] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

**Comment by AndroidBob** [comment]
Build [1](https://10.10.124.134/job/wire-android-reloaded/job/PR-1852/1/) **failed**.

File: app/src/main/kotlin/com/wire/android/ui/authentication/login/sso/LoginSSOViewModel.kt [hunk 1]
```diff
@@ -90,10 +91,48 @@ class LoginSSOViewModel @Inject constructor(
...
+                }
+
+                authScope.ssoLoginScope.fetchSSOSettings().also {
+                    when (it) {
+                        is FetchSSOSettingsUseCase.Result.Failure -> {}
+                        is FetchSSOSettingsUseCase.Result.Success -> {
+                            it.defaultSSOCode?.let { ssoCode ->
+                                val ssoCodeWithPrefix = ssoCode.startsWith("wire-").let { hasPrefix ->
+                                    if (hasPrefix) ssoCode else "wire-$ssoCode"
+                                }
```
Comment (alexandreferris) [inline app/src/main/kotlin/com/wire/android/ui/authentication/login/sso/LoginSSOViewModel.kt]: Could it be improved a little bit like this? then we don't need the `.let { .. }` scope 🤔 

```suggestion
                                val ssoCodeWithPrefix = if (ssoCode.startsWith("wire-")) ssoCode else "wire-$ssoCode"
```

**Review by alexandreferris — APPROVED** [review]
Looking good 👌🏻 
Just a tiny comment on code improvement but it also ain't that much necessary, but just in case xD

**Comment by AndroidBob** [comment]
Build [2](https://10.10.124.134/job/wire-android-reloaded/job/PR-1852/2/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv4.2.0(6759096)(6759096)-staging-compat.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/staging/compat/com.wire.android-vv4.2.0(6759096)(6759096)-staging-compat.apk)

**Review by yamilmedina — APPROVED** [review]

**Pull request merged**